demisto_sdk/dev_tools/lint_manager.py

```diff
@@ -0,0 +1,139 @@
+from demisto_sdk.common.tools import print_color, LOG_COLORS
+
+from demisto_sdk.dev_tools.linter import Linter
+
+
+class LintManager:
+    def __init__(self, project_dir: str, no_test: bool = False, no_pylint: bool = False, no_flake8: bool = False,
+                 no_mypy: bool = False, verbose: bool = False, root: bool = False, keep_container: bool = False,
+                 cpu_num: int = 0, parallel: bool = False, configuration: Configuration = Configuration()):
+
+        if no_test and no_pylint and no_flake8 and no_mypy:
+            raise ValueError("Nothing to run as all --no-* options specified.")
+
+        self.parallel = parallel
+        self.log_verbose = verbose
+        self.root = root
+        self.parallel = parallel
+        self.keep_container = keep_container
+        self.cpu_num = cpu_num
+        self.common_server_created = False
+        self.run_args = {
+            'pylint': not no_pylint,
+            'flake8': not no_flake8,
+            'mypy': not no_mypy,
+            'tests': not no_test
+        }
+        self.pkgs = project_dir.split(',')
+        self.configuration = configuration
+
+    def run_dev_packages(self) -> int:
+        # if we dont run in parallel we run a single process
+        if not self.parallel:
+            linter = Linter(self.pkgs[0], no_test=not self.run_args['tests'],
+                            no_pylint=not self.run_args['pylint'], no_flake8=not self.run_args['flake8'],
+                            no_mypy=not self.run_args['mypy'], verbose=self.log_verbose, root=self.root,
+                            keep_container=self.keep_container, cpu_num=self.cpu_num, configuration=self.configuration)
+
+            return linter.run_dev_packages()
+
+        # we run parallel processes
+        else:
+            max_workers = int(os.getenv("MAX_WORKERS", "10"))
+            pkg_dirs = self.pkgs
+            pkgs_to_run = []
+            for dir in pkg_dirs:
+                if self._should_run_pkg(pkg_dir=dir):
+                    pkgs_to_run.append(dir)
+
+            print("Starting parallel run for [{}] packages with [{}] max workers".format(len(pkgs_to_run), max_workers))
+            fail_pkgs = []
+            good_pkgs = []
+
+            # run CommonServer non parallel to avoid conflicts
+            # when we modify the file for mypy includes
+            if 'Scripts/CommonServerPython' in pkgs_to_run:
+                pkgs_to_run.remove('Scripts/CommonServerPython')
+                res, _ = self._run_single_package_thread(package_dir='Scripts/CommonServerPython')
+                if res == 0:
+                    good_pkgs.append('Scripts/CommonServerPython')
+
+                else:
+                    fail_pkgs.append('Scripts/CommonServerPython')
+
+            with concurrent.futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
+                futures_submit = [executor.submit(self._run_single_package_thread, dir) for dir in pkgs_to_run]
```

Comment:
dir is a python saved word, lets rename this arg

Reply:
changed